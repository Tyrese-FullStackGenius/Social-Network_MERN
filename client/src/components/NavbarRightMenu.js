import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import Button from '@material-ui/core/Button';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import { withStyles } from '@material-ui/core/styles';

const styles = {
  link: {
    outline: 'none',
    textDecoration: 'none'
  },
  menuButton: {
    color: '#fff',
    fontSize: '18px',
    marginRight: '-15px',
    textTransform: 'none'
  }
};

class NavbarRightMenu extends Component {
  state = {
    anchorEl: null
  };

  handleClick = (event) => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  render() {
    const { classes, logoutUser, user } = this.props;
    const { anchorEl } = this.state;

    return (
      <div>
        <Button
          aria-owns={anchorEl ? 'right-menu' : null}
          aria-haspopup="true"
          className={classes.menuButton}
          onClick={this.handleClick}
        >
          {user.name}
        </Button>
        <Menu
          id="right-menu"
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={this.handleClose}
        >
          <Link className={classes.link} to={`/profile/${user.userId}`}>
            <MenuItem onClick={this.handleClose}>Profile</MenuItem>
          </Link>
          <Link className={classes.link} to="/discover">
            <MenuItem onClick={this.handleClose}>Discover</MenuItem>
          </Link>
          <MenuItem onClick={logoutUser}>Logout</MenuItem>
        </Menu>
      </div>
    );
  }
}

NavbarRightMenu.propTypes = {
  classes: PropTypes.object.isRequired,
  logoutUser: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired
};

export default withStyles(styles)(NavbarRightMenu);
