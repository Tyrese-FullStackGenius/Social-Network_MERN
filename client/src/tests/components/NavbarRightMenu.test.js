import React from 'react';
import { shallow } from 'enzyme';
import NavbarRightMenu from '../../components/NavbarRightMenu';

it('should render NavbarLeftMenu', () => {
  const wrapper = shallow(<NavbarRightMenu logoutUser={jest.fn()} user={{}} />);
  expect(wrapper).toMatchSnapshot();
});
