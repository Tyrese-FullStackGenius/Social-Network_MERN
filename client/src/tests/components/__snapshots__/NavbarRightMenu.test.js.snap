// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NavbarLeftMenu 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(NavbarRightMenu)
    logoutUser={[Function]}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "link": "NavbarRightMenu-link-1",
        "menuButton": "NavbarRightMenu-menuButton-2",
      },
      "logoutUser": [Function],
      "user": Object {},
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "link": "NavbarRightMenu-link-1",
          "menuButton": "NavbarRightMenu-menuButton-2",
        },
        "logoutUser": [Function],
        "user": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
